juju/juju
# refactor: wire in model service bootstrap worker

This commit wires in the model service(s) to the bootstrap worker replacing calls to Mongo state for model information. This allows us to keep transitioning off of Mongo and over to dqlite.
    
This change is functionally equivalent with the source of model information being changed.

This work is being done as part of bootstrapping authorised keys. We required the model service(s) to perform this function correctly.

As part of this PR we have also removed manual environ creation using domain services and opted to use the provider tracker available in the manifold.

## Checklist


## QA steps

1. Run unit tests for `internal/worker/bootstrap`
2. Bootstrap a local controller and switch to the controller model. Confirm the controller charm is health and deployed with reported agent versions.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-6478
